src/wallet/wallet2.cpp

```diff
@@ -1332,65 +1323,34 @@ void wallet2::process_new_transaction(const crypto::hash &txid, const cryptonote
               << " from received " << print_money(tx.vout[o].amount) << " output already exists with "
               << print_money(m_transfers[kit->second].amount()) << ", replacing with new output");
-// monero version
-/*
-	  LOG_ERROR("Public key " << epee::string_tools::pod_to_hex(kit->first)
-              << " from received " << print_money(tx_scan_info[o].amount) << " output already exists with "
-              << print_money(m_transfers[kit->second].amount()) << ", replacing with new output");
-          // The new larger output replaced a previous smaller one
-*/
 
           // The new larger output replaced a previous smaller one
-	  /*
           auto iter = std::find_if(
               tx_money_got_in_outs.begin(),
               tx_money_got_in_outs.end(),
               [&tx_scan_info,&tx,&o](const tx_money_got_in_out& value)
               {
                 return value.index == tx_scan_info[o].received->index &&
-                  value.amount == tx.vout[o].amount;
+                       value.amount == tx_scan_info[o].amount;
               }
             );
-          if (iter == tx_money_got_in_outs.end())
-            LOG_ERROR("Could not find the output we just added, this should never happen");
-          else
-            tx_money_got_in_outs.erase(iter);
-	  */
-	  // Monero version of the fix - 9/25/18 rtharp
-          auto iter = std::find_if(
-              tx_money_got_in_outs.begin(),
-              tx_money_got_in_outs.end(),
-              [&tx_scan_info,&tx,&o](const tx_money_got_in_out& value)
-              {
-                return value.index == tx_scan_info[o].received->index &&
-                  value.amount == tx_scan_info[o].amount;
-              }
-            );
-          if (iter == tx_money_got_in_outs.end())
-          {
-            THROW_WALLET_EXCEPTION_IF(1,
-                error::wallet_internal_error, "Could not find the output we just added, this should never happen");
-          }
 
-          // Doyle said to remove - rtharp.
-          //THROW_WALLET_EXCEPTION_IF(iter->amount < tx_scan_info[o].amount,
-          //    error::wallet_internal_error, "Unexpected values of new and old outputs");
-          THROW_WALLET_EXCEPTION_IF(m_transfers[kit->second].amount() > tx_scan_info[o].amount,
-              error::wallet_internal_error, "Unexpected values of new and old outputs");
+          // Monero fix - 9/25/18 rtharp, doyle
+          THROW_WALLET_EXCEPTION_IF(iter == tx_money_got_in_outs.end(), error::wallet_internal_error, "Could not find the output we just added, this should never happen");
+          THROW_WALLET_EXCEPTION_IF(m_transfers[kit->second].amount() > iter->amount, error::wallet_internal_error, "Unexpected values of new and old outputs");
           iter->amount -= m_transfers[kit->second].amount();
-          //tx_money_got_in_outs[tx_scan_info[o].received->index] -= tx_scan_info[o].amount;
+
+          if (iter->amount == 0)
```

Comment:
Since we don't add up the amounts, this must always be 0 in our case, right?

Reply:
This is the extra amount. The (incoming output > the old output) we have in our wallet. So we need to only leave the remainder to be added to the wallet as a payment_detail. And actually, now thinking about it, this must always be greater than 0 in that case.

We could throw an exception here if it's not. Thoughts?